common/neon/neon_tx_error_parser.py

```diff
@@ -0,0 +1,116 @@
+    def check_if_already_finalized(self) -> bool:
+        log_list = self._get_evm_error_log_list()
+        return any(log_rec.code == NeonTxErrorLogInfo.ErrorCode.StorageAccountFinalized for log_rec in log_list)
+
+    @cached_method
+    def get_nonce_error(self) -> tuple[int, int] | None:
+        log_list = self._get_evm_error_log_list()
+        for log_rec in log_list:
+            if log_rec.code == NeonTxErrorLogInfo.ErrorCode.InvalidTransactionNonce:
+                address = log_rec.data[0:19]
+                state_tx_cnt = int.from_bytes(data[20:27])
+                tx_nonce = int.from_bytes(data[28:35])
+                return int(state_tx_cnt), int(tx_nonce)
+        return None
+
+    @cached_method
+    def get_out_of_gas_error(self) -> tuple[int, int] | None:
+        log_list = self._get_evm_error_log_list()
+        for log_rec in log_list:
+            if log_rec.code == NeonTxErrorLogInfo.ErrorCode.OutOfGas:
+                has_gas_limit = int.from_bytes(log_rec.data[0:31])
+                req_gas_limit = int.from_bytes(log_recdata[32:64])
+                return int(has_gas_limit), int(req_gas_limit)
+        return None
+
+    @cached_method
+    def _get_evm_log_list(self) -> tuple[str, ...]:
+        if isinstance(self._receipt, SolRpcSendTxErrorInfo):
+            rpc_meta = self._receipt
+        elif isinstance(self._receipt, SolRpcTxSlotInfo):
+            rpc_meta = self._receipt.transaction.meta
+        else:
+            return tuple()
+
+        log_list: list[str] = list()
+        log_state = SolTxLogTreeDecoder.decode(self._tx.message, rpc_meta, self._tx.account_key_list)
+        for log_info in log_state.log_list:
+            if log_info.prog_id == NeonProg.ID:
+                log_list.extend(log_info.log_msg_list())
+            for inner_log_info in log_info.inner_log_list:
+                if inner_log_info.prog_id == NeonProg.ID:
+                    log_list.extend(inner_log_info.log_msg_list())
+        return tuple(log_list)
+
+    @cached_method
+    def _get_evm_error_log_list(self) -> tuple[NeonTxErrorLogInfo, ...]:
+        if isinstance(self._receipt, SolRpcSendTxErrorInfo):
+            rpc_meta = self._receipt
+        elif isinstance(self._receipt, SolRpcTxSlotInfo):
+            rpc_meta = self._receipt.transaction.meta
+        else:
+            return tuple()
+
+        log_list: list[str] = list()
+        log_state = SolTxLogTreeDecoder.decode(self._tx.message, rpc_meta, self._tx.account_key_list)
+
+        sol_tx_ix_meta_info = SolTxIxMetaInfo(
+            sol_tx_sig=SolTxSig.default(),
+            slot=0,
+            sol_ix_idx=1,
+            sol_inner_ix_idx=None,
+            is_success=False,
+            _rpc_tx_ix=None,
+            _tx_acct_key_list=None)
+
+        error_log_list: list[NeonTxErrorLogInfo] = list()
+        for log_info in log_state.log_list:
+            if log_info.prog_id == NeonProg.ID:
+                log_list.extend(log_info.log_msg_list())
```

Comment:
Just replace this copy-paste with the call `evm_log_list = self._get_evm_log_list()`